src/Shared/NativeMethodsShared.cs

```diff
@@ -570,6 +642,98 @@ internal static int SetErrorMode(int newMode)
         internal static extern void GetNativeSystemInfo(ref SYSTEM_INFO lpSystemInfo);
 
         /// <summary>
+        /// Convert SYSTEM_INFO architecture values to the internal enum
+        /// </summary>
+        /// <param name="arch"></param>
+        /// <returns></returns>
+        private static ProcessorArchitectures ConvertSystemArcitecture(ushort arch)
+        {
+            switch (arch)
+            {
+                case PROCESSOR_ARCHITECTURE_INTEL:
+                    return ProcessorArchitectures.Architecture_X86;
+                case PROCESSOR_ARCHITECTURE_AMD64:
+                    return ProcessorArchitectures.Architecture_X64;
+                case PROCESSOR_ARCHITECTURE_ARM:
+                    return ProcessorArchitectures.Architecture_ARM;
+                case PROCESSOR_ARCHITECTURE_IA64:
+                    return ProcessorArchitectures.Architecture_IA64;
+                default:
+                    return ProcessorArchitectures.Architecture_Unknown;
+            }
+        }
+
+        /// <summary>
+        /// Read system info values
+        /// </summary>
+        private static void ReadSystemInfo()
+        {
+            if (IsWindows)
+            {
+                SYSTEM_INFO systemInfo = new NativeMethodsShared.SYSTEM_INFO();
+
+                GetSystemInfo(ref systemInfo);
+
+                s_processorArchitecture = ConvertSystemArcitecture(systemInfo.wProcessorArchitecture);
+
+                GetNativeSystemInfo(ref systemInfo);
+                s_processorArchitectureNative = ConvertSystemArcitecture(systemInfo.wProcessorArchitecture);
+
+                s_processorArchitectureSet = true;
+            }
+            else
+            {
+                try
+                {
+                    // On Unix run 'uname -m' to get the architecture. It's common for
+                    // Linux and Mac
+                    using (
+                        var proc =
+                            System.Diagnostics.Process.Start(
+                                new System.Diagnostics.ProcessStartInfo("uname")
+                                    {
+                                        Arguments = "-m",
+                                        UseShellExecute = false,
+                                        RedirectStandardOutput = true,
+                                        CreateNoWindow = true
+                                    }))
+                    {
+                        proc.WaitForExit(1000);
```

Comment:
Well, I needed some magic:hammer:. The function is just a kernel property wrapper, so should be fast. I chose 1 sec, but whatever the number, it would be magic. Any suggestions?